library/Icinga/Util/Csp.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+namespace Icinga\Util;
+
+use Icinga\Web\Response;
+use Icinga\Web\Session;
+use Icinga\Web\Window;
+
+class Csp
+{
+    /** @var self */
+    private static $instance = null;
+
+    /** @var string */
+    private $styleNonce = null;
+
+    /** @var string */
+    private $scriptNonce = null;
+
+    private function loadNoncesFromSession()
+    {
+        $instance = self::getInstance();
+        $windowSession = Window::getInstance()->getSessionNamespace('csp');
+
+        $sessionStyleNonce = $windowSession->get('csp_style_nonce');
+        $sessionScriptNonce = $windowSession->get('csp_script_nonce');
+
+        if (! empty($sessionStyleNonce)) {
```

Comment:
Please remove the empty check. This may hide bugs. `loadNoncesFromSession()` **must** apply the session state. If the nonces are null, they have to be set to null.